koku/api/iam/view/user_preference.py

```diff
@@ -76,11 +76,15 @@ def create(self, request, *args, **kwargs):
 
         @apiParam {String} user_uuid User unique ID.
 
-        @apiParam (Request Body) {String} preference a stringified JSON object
+        @apiParam (Request Body) {String} name User preference name.
+                                               No requirement to match the key in preference
+        @apiParam (Request Body) {Object} preference Preference value
+        @apiParam (Request Body) {String} [description] The description of the user preference.
         @apiParamExample {json} Request Body:
             {
-                "name": "my-preference-name",
+                'name': 'my-preference-name',
```

Comment:
Should be double quotes to be valid JSON.

```
"name": "my-preference-name",
"preference": {"currency": "USD"},
"description": "description of my-preference-name"
```

Reply:
Good catch. updated